agent/submatview/materializer.go

```diff
@@ -80,6 +79,18 @@ func NewMaterializer(deps Deps) *Materializer {
 		retryWaiter: deps.Waiter,
 		updateCh:    make(chan struct{}),
 	}
+	if deps.Waiter == nil {
```

Comment:
```suggestion
	if v.retryWaiter == nil {
```
Technically the same thing, but this one seems less confusing should this ever get refactored and additional lines be added in between the field initial assignment and the later defaulting.